system/jlib/jfile.cpp

```diff
@@ -2027,18 +2042,24 @@ void CFileIO::close()
 #ifdef CFILEIOTRACE
         DBGLOG("CFileIO::close(%d), extraFlags = %d", tmpHandle, extraFlags);
 #endif
-        if (extraFlags & IFEnocache)
+        if (extraFlags & (IFEnocache | IFEsync))
         {
             if (openmode != IFOread)
             {
+                if (extraFlags & IFEsync)
```

Comment:
All of this code to line 2061 could probably move into a common function to avoid the duplication and differences.

Reply:
done